### PR TITLE
Clarify usage of optional outputs in output tuples

### PR DESCRIPTION
**Pull request by pinin4fjords**
From conversation in https://nfcore.slack.com/archives/CE4K7FEHE/p1669023424969749, I'm proposing a clarification to the associated documentation regarding optional outputs in otherwise non-optional tuples. 

<a href="https://gitpod.io/#https://github.com/nf-core/nf-co.re/pull/1461"></a>

### CODE REVIEW

**Comment by pinin4fjords** [comment]
@nf-core-bot fix linting

File: markdown/developers/modules.md [hunk 1]
```diff
@@ -510,6 +510,14 @@ process {
 
 3. Optional inputs are not currently supported by Nextflow. However, passing an empty list (`[]`) instead of a file as a module parameter can be used to work around this issue.
 
+4. Optional outputs SHOULD be marked as optional:
+
+   ```nextflow
+   tuple val(meta), path('*.tab')                   , optional:true, emit: tab
+   ```
+
+5. Since it is not currently possible to mark individual elements of a tuple as optional, optional outputs SHOULD NOT be placed in output channels with mandatory outputs.
```
Comment (mahesh-panchal) [inline markdown/developers/modules.md:519]:
```suggestion
5. Each output file SHOULD be emitted in it's own channel, along with the `meta` map if provided ( the exception is the versions.yml ).
```

Comment (mahesh-panchal) [reply markdown/developers/modules.md:519]: @drpatelh @jfy133 This is usually enforced right? Each file should be in it's own channel?

Comment (pinin4fjords) [reply markdown/developers/modules.md:519]: Ahh, that's new on me, I have a couple of modules with multiple-file output tuples

Comment (jfy133) [reply markdown/developers/modules.md:519]: yes, we enforce that - as we don't want to make assumptions what people want to use each output file from (what you may consider should 'travel together' may not be what others do...). So yes, lots of re-joining tuples...

File: markdown/developers/modules.md [hunk 2]
```diff
@@ -510,6 +510,14 @@ process {
 
 3. Optional inputs are not currently supported by Nextflow. However, passing an empty list (`[]`) instead of a file as a module parameter can be used to work around this issue.
 
+4. Optional outputs SHOULD be marked as optional:
+
+   ```nextflow
+   tuple val(meta), path('*.tab')                   , optional:true, emit: tab
```
Comment (mahesh-panchal) [inline markdown/developers/modules.md:516]:
```suggestion
   tuple val(meta), path('*.tab'), emit: tab,  optional: true
```
This is just a style suggestion. I prefer the emits to line up and spaces

**Review by mahesh-panchal — APPROVED** [review]

File: markdown/developers/modules.md (hunk 2 above)
Comment (pinin4fjords) [reply markdown/developers/modules.md:516]: Sure, I just copy/pasted directly from a module, but happy to adjust.

**Pull request merged**